packages/simcore-sdk/tests/integration/test_node_data_data_manager.py

```diff
@@ -0,0 +1,201 @@
+# pylint:disable=unused-variable
+# pylint:disable=unused-argument
+# pylint:disable=redefined-outer-name
+# pylint:disable=too-many-arguments
+
+import hashlib
+import os
+from asyncio import BaseEventLoop
+from pathlib import Path
+from typing import Callable, Set, Tuple
+from uuid import uuid4
+
+import pytest
+from simcore_sdk.node_data import data_manager
+
+pytest_simcore_core_services_selection = ["postgres", "storage"]
+
+pytest_simcore_ops_services_selection = ["minio", "adminer"]
+
+
+# UTILS
+
+
+def _remove_file_or_folder(file_or_folder: Path) -> None:
+    if file_or_folder.is_file():
+        file_or_folder.unlink()
+        assert file_or_folder.exists() is False
+        file_or_folder.touch()
+        assert file_or_folder.exists() is True
+    else:
+        os.system(f"rm -rf {file_or_folder}")
+        assert file_or_folder.exists() is False
+        file_or_folder.mkdir(parents=True, exist_ok=True)
+        assert file_or_folder.exists() is True
+
+
+def _path_hashes(path_to_hash: Path) -> Set[Tuple[str, str]]:
+    def _hash_path(path: Path):
+        sha256_hash = hashlib.sha256()
+        with open(path, "rb") as f:
+            # Read and update hash string value in blocks of 4K
+            for byte_block in iter(lambda: f.read(4096), b""):
+                sha256_hash.update(byte_block)
+        return sha256_hash.hexdigest()
+
+    def _path_from_root(root_path: Path, full_path: Path) -> str:
+        return str(full_path).replace(str(root_path), "")

```

Comment:
TIP: you want a relative path here, right? this is simply (and more readable) as:
```python

return { (full_path.relative_to(root_path), compute_hash(path_to_hash)) }
```